src/RecordCommand.cc

```diff
@@ -623,9 +624,11 @@ static void copy_preload_sources_to_trace(const string& trace_dir) {
     FATAL() << "Can't spawn 'zip'";
   }
   posix_spawn_file_actions_destroy(&actions);
-  int status;
-  waitpid(pid, &status, 0);
-  LOG(info) << "Got zip status " << WaitStatus(status);
+  WaitResult result = WaitManager::wait_exit(WaitOptions(pid));
```

Comment:
Does converting these ancillary waits that don't have anything to do with the tracee actually accomplish anything?

Reply:
The only reason to do this is to make all `wait`-related syscalls go through the WaitManager. I think that's simpler than using naked wait syscalls for some processes and WaitManager for others. Mixing those for the same process would be risky because you might accidentally bypass WaitManager's stashed stops.

Reply:
Well, if that's the case, you missed one in ExportImportCheckpoints.cc

Reply:
Thanks --- e231941fa978e73f29a2a3a346e2892453ac767e